router/docs/Configuration.md

```diff
@@ -0,0 +1,101 @@
+# Router Configuration Conventions
+
+This document outlines the conventions used in Cosmo's configuration schema.
+
+## General Structure
+
+- Configuration is defined in a JSON Schema format
+- All configuration options have specific types with validation constraints
+- Properties use `snake_case` naming convention
+- Nested objects are used to group related configuration options
+
+## Validation Rules
+
+- Required fields are explicitly marked with `required` arrays
+- `additionalProperties: false` ensures no unknown properties are allowed
+- Minimum/maximum constraints are specified where applicable
+- Enums are used for properties with a fixed set of allowed values
+
+## Common Patterns
+
+- Property names use nouns or configurations, **not** verb-prefixed names (use `storage` not `get_storage` or `use_storage`)
+- Every configurable component uses an `enabled` boolean property to toggle features without removing them from the configuration
+- URLs follow the format `scheme://host:port`
+- File paths use the `format: file-path` validator
+- Sizes are specified using string format with units (e.g., "100MB")
+- Time durations use string format with units (e.g., "1s", "1m", "1h")
+- Default values are specified in `pkg/config/config.go` via struct tags.
+
+## Documentation
+
+- Every property includes a `description` field explaining its purpose and usage
+- Deprecated properties are marked with `deprecated` and should include a reason for deprecation
+
+# Contributing New Configuration Options
+
+To add a new configuration option, you need to update two files:
+
+## 1. Update `pkg/config/config.go`
+
+First, add the new option to the appropriate struct in `pkg/config/config.go`:
+
+```go
+type YourFeatureConfig struct {
+    Enabled bool   `yaml:"enabled" envDefault:"false" env:"YOUR_FEATURE_ENABLED"`
+    Timeout string `yaml:"timeout" envDefault:"30s" env:"YOUR_FEATURE_TIMEOUT"`
+}
+
+type Config struct {
+    // ... existing fields
+    YourFeature YourFeatureConfig `yaml:"your_feature,omitempty"`
+}
+```
+
+The struct field tags define:
+- `yaml`: The field name in YAML configuration
+- `envDefault`: Default value for the field
+- `env`: Environment variable name for this option
+- `omitempty`: Makes the field optional in the YAML config
```

Comment:
```suggestion
- `omitempty`: Makes the field optional in the YAML config
- `envPrefix`: Can be used in a field that is a complex type to set a prefix to all environment variables used in it
```

Reply:
This will be in a bad order if you accept the change as-is, perhaps this should be two ULs